sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/conditionals.scala

```diff
@@ -296,9 +295,7 @@ case class CaseKeyWhen(key: Expression, branches: Seq[Expression]) extends CaseW
   }
 
   private def equalNullSafe(l: Any, r: Any) = {
```

Comment:
@marmbrus 
Shall I rename the function   private def equalNullSafe(l: Any, r: Any) 
to  private def equal(l: Any, r: Any) ?